src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -450,13 +450,18 @@ public UserImpl createUser(DataObject user)
             ? new User.Profile(id, user.getString("banner", null), user.getInt("accent_color", User.DEFAULT_ACCENT_COLOR_RAW))
             : null;
 
+        User.AvatarDecoration avatarDecoration = user.optObject("avatar_decoration_data")
+                .map(o -> new User.AvatarDecoration(o.getString("asset"), o.getString("sku_id")))
+                .orElse(null);
+
         if (newUser)
         {
             // Initial creation
             userObj.setName(user.getString("username"))
                    .setGlobalName(user.getString("global_name", null))
                    .setDiscriminator(Short.parseShort(user.getString("discriminator", "0")))
                    .setAvatarId(user.getString("avatar", null))
+                   .setAvatarDecoration(avatarDecoration)
```

Comment:
What if this is not a new user?

Reply:
Wym? Should I also fire user update event with this property?

Reply:
It should be treated like all the other properties. The `updateUser` method called in the `else` branch is used for those cases.

Reply:
Do I create an event for it then? Similar to UserUpdateAvatarEvent

Reply:
Yes, please add an event.

Reply:
Should be good, tell me if I did something wrong